src/components/EmojiPicker/EmojiPickerMenu/index.native.js

```diff
@@ -1,79 +1,53 @@
-    onEmojiSelected: PropTypes.func.isRequired,
+const propTypes = emojiPickerMenuPropTypes;
 
-    /** Stores user's preferred skin tone */
-    preferredSkinTone: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+const defaultProps = emojiPickerMenuDefaultProps;
 
-    /** Stores user's frequently used emojis */
-    // eslint-disable-next-line react/forbid-prop-types
-    frequentlyUsedEmojis: PropTypes.arrayOf(PropTypes.object),
-
-    /** Props related to translation */
-    ...withLocalizePropTypes,
-};
-
-const defaultProps = {
-    preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
-    frequentlyUsedEmojis: [],
-};
-
-function EmojiPickerMenu({preferredLocale, onEmojiSelected, preferredSkinTone, translate, frequentlyUsedEmojis}) {
+function EmojiPickerMenu({onEmojiSelected}) {
     const styles = useThemeStyles();
-    const StyleUtils = useStyleUtils();
-    const emojiList = useAnimatedRef();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    const allEmojis = useMemo(() => EmojiUtils.mergeEmojisWithFrequentlyUsedEmojis(emojis), [frequentlyUsedEmojis]);
-    const headerEmojis = useMemo(() => EmojiUtils.getHeaderEmojis(allEmojis), [allEmojis]);
-    const headerRowIndices = useMemo(() => _.map(headerEmojis, (headerEmoji) => Math.floor(headerEmoji.index / CONST.EMOJI_NUM_PER_ROW)), [headerEmojis]);
-    const [filteredEmojis, setFilteredEmojis] = useState(allEmojis);
-    const [headerIndices, setHeaderIndices] = useState(headerRowIndices);
     const {windowWidth} = useWindowDimensions();
+    const {translate} = useLocalize();
+    const {
+        allEmojis,
+        headerEmojis,
+        headerRowIndices,
+        filteredEmojis,
+        headerIndices,
+        setFilteredEmojis,
+        setHeaderIndices,
+        isListFiltered,
+        suggestEmojis,
+        preferredSkinTone,
+        listStyle,
+        emojiListRef,
+    } = useEmojiPickerMenu();
     const {singleExecution} = useSingleExecution();
 
-    useEffect(() => {
-        setFilteredEmojis(allEmojis);
-    }, [allEmojis]);
-
-    useEffect(() => {
-        setHeaderIndices(headerRowIndices);
-    }, [headerRowIndices]);
-
-    const getItemLayout = (data, index) => ({length: CONST.EMOJI_PICKER_ITEM_HEIGHT, offset: CONST.EMOJI_PICKER_ITEM_HEIGHT * index, index});
-
     /**
      * Filter the entire list of emojis to only emojis that have the search term in their keywords
      *
      * @param {String} searchTerm
      */
     const filterEmojis = _.debounce((searchTerm) => {
-        const normalizedSearchTerm = searchTerm.toLowerCase().trim().replaceAll(':', '');
+        const [normalizedSearchTerm, newFilteredEmojiList] = suggestEmojis(searchTerm);
 
-        if (emojiList.current) {
-            emojiList.current.scrollToOffset({offset: 0, animated: false});
+        if (emojiListRef.current) {
+            emojiListRef.current.scrollToOffset({offset: 0, animated: false});
```

Comment:
I think we can use `scrollTo` here, same as we do in `index.js`

Reply:
I'm noticing a small flickering on mobile here when I am using `scrollTo` - it looks like when using it, the scroll to top is faster than filtering the list as it's not animating - list is scrolled to top first and after a few ms data is overwritten with new data. Not sure why, but it works correct when using the approach above. Attaching a video of the issue when using `scrollTo` here.

https://github.com/Expensify/App/assets/13985840/ee9bfe88-0ba1-40f4-99ad-3b49a5363185